core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -109,7 +110,9 @@ protected String blobName(String name) {
     }
 
     protected T read(BytesReference bytes) throws IOException {
-        try (XContentParser parser = XContentHelper.createParser(namedXContentRegistry, bytes)) {
+        // UNSUPPORTED_OPERATION_DEPRECATION_HANDLER is fine because we don't have deprecated fields in the blob store
```

Comment:
Oh boy. I'm not sure what the deprecated field *does* then..... I can replace this with something else, but the `LOGGING_DEPRECATION_HANDLER` feels pretty weird here. We'd stuff that warning off in the `ThreadLocal` and if we're lucky we send it back to the user and they get confused because it points them to fields that they didn't send. If we're not lucky it sits in the ThreadLocal forever with no one to look at it.

Reply:
I think we can either ignore it or fix BlobStoreIndexShardSnapshot to not have any deprecated fields.